hikari/channels.py

```diff
@@ -95,6 +97,95 @@ def __str__(self) -> str:
         return self.name
 
 
+@attr_extensions.with_copy
+@attr.s(eq=True, hash=True, init=True, kw_only=True, slots=True, weakref_slot=False)
+class ChannelFollow:
+    """Represents a channel follow."""
+
+    app: traits.RESTAware = attr.ib(repr=False, eq=False, hash=False, metadata={attr_extensions.SKIP_DEEP_COPY: True})
+    """The client application that models may use for procedures."""
```

Comment:
```suggestion
    """Return the app instance for this application.
    
    Returns
    -------
    hikari.traits.RESTAware
        The REST-aware application object.
    """
```
... no, we haven't done this before, but iirc there is an issue to start doing this. Partially because it allows each object in pdoc to have a properly documented hyperlink to the type even when the type hints mess up on the object preview, and also due to how pdoc interprets the `attrs` conduit, it will document these fields as properties.

Pretending these are properties also encourages the user to not try and change them manually, so I'd like to start enforcing this everywhere. The `hikari.events` module is already using this format for everything.

![image](https://user-images.githubusercontent.com/69713762/92594034-1803a980-f29a-11ea-9891-307952622499.png)

You can see a preview of this [here](https://nekokatt.github.io/hikari/hikari/events/guild_events.html#hikari.events.guild_events.BanCreateEvent.app) to see how this looks in more detail.

Reply:
The pdoc bug is caused by inheritance of abstract classes, not really sure why, but I can tell you that :)

Reply:
#148